implementation/src/main/java/io/smallrye/jwt/JsonUtils.java

```diff
@@ -0,0 +1,87 @@
+package io.smallrye.jwt;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonValue;
+
```

Comment:
The `JsonUtils` class contains several Json wrapping methods previously duplicated elsewhere in the implementation.

Reply:
Please make sure the code from `JWTCallerPrincipal` is moved to this class, not the one from `CommonJwtProducer` (which indeed can be dropped)

Reply:
The version from `JWTCallerPrincipal` is what was used. Note that the only significant difference between the two is that iteration of a `Collection` will add `JsonValue.NULL` if the entry is `null`. Also note that the method of creating Json primitives using a `JsonObjectBuilder` and a "tmp" key was replaced with `Json.createValue` factory calls.

Reply:
Re `JsonValue.NULL` - lets deal with it as part of a separate issue (if there will be a specific regression) - the concerns around replacing `null` everywhere was discussed as part of #103.